src/ThreeEditor/components/Sidebar/tree/SidebarTreeItem.tsx

```diff
@@ -0,0 +1,226 @@
+	};
+}
+
+function isHidable(object: Object3D | ISimulationObject) {
+	if ('notHidable' in object) {
+		return !object.notHidable;
+	}
+	return true;
+}
+
+export function SidebarTreeItem(props: {
+	treeRef: TreeMethods | null;
+	objectRefs: React.MutableRefObject<Map<string, HTMLDivElement>>;
+	node: NodeModel<{
+		object: Object3D | SimulationObject3D;
+	}>;
+	depth: number;
+	isOpen: boolean;
+	onToggle: () => void;
+	editor: Editor;
+}) {
+	const { objectRefs, node, depth, isOpen, onToggle, editor, treeRef } = props;
+	const inputRef = useRef<HTMLInputElement>(null);
+
+	const theme = useTheme();
+
+	const object = node.data!.object;
+
+	const { state: watchedObject } = useSmartWatchEditorState(editor, object);
+
+	const [mode, setMode] = useState<'view' | 'edit'>('view');
+
+	const popupState = usePopupState({ variant: 'popover', popupId: 'sidebarTreeItemMenu' });
+
+	const contextOptions = (() => {
+		const options: JSX.Element[] = [];
+
+		if (canChangeName(object)) {
+			options.push(
+				<MenuItem
+					onClick={() => {
+						popupState.close();
+						setMode('edit');
+					}}>
+					Rename
+				</MenuItem>
+			);
+		}
+
+		if (isOutput(object))
+			options.push(
+				<MenuItem
+					onClick={() => {
+						editor.execute(new AddQuantityCommand(editor, object));
+						if (!isOpen) onToggle();
+						popupState.close();
+					}}>
+					Add Quantity
+				</MenuItem>
+			);
+
+		if (isRemovable(object))
+			options.push(
+				<MenuItem
+					onClick={() => {
+						editor.execute(getRemoveCommand(editor, object));
+						popupState.close();
+					}}>
+					Delete
+				</MenuItem>
+			);
+
+		return options;
+	})();
+
+	useEffect(() => {
+		if (mode === 'edit') {
+			inputRef?.current?.focus();
+		}
+	}, [mode]);
+
+	const onObjectAdded = useCallback(
+		(newObject: Object3D) => {
+			if (newObject.parent === object) {
+				treeRef?.open(node.id);
+			}
+		},
+		[node.id, object, treeRef]
+	);
+
+	useSignal(editor, 'objectAdded', onObjectAdded);
+
+	return (
+		<>
+			<Box
+				ref={ref => objectRefs.current.set(object.uuid, ref! as HTMLDivElement)}
+				sx={{
+					'marginLeft': theme.spacing(depth * 2.5),
+
+					'display': 'flex',
+					'flexDirection': 'row',
+					'alignItems': 'center',
+					'paddingLeft': theme.spacing(0.5),
+					':hover': {
+						backgroundColor: theme.palette.action.hover
+					}
+				}}>
+				{object.children.length > 0 &&
+					!(object as SimulationObject3D).notVisibleChildren && (
+						<ChevronRightIcon
+							onClick={onToggle}
+							sx={{
+								transform: isOpen ? 'rotate(90deg)' : 'rotate(0deg)',
+								transition: 'transform 0.2s'
+							}}
+						/>
+					)}
+				<Stack
+					direction='row'
+					onClick={() => editor.selectById(object.id)}
+					width='100%'
+					sx={{ cursor: 'pointer' }}
+					{...bindContextMenu(popupState)}>
+					<Typography
+						onKeyDown={ev => {
+							if (ev.key === 'F2') {
+								ev.preventDefault();
+								setMode('edit');
+							}
+						}}
```

Comment:
![image](https://user-images.githubusercontent.com/26445545/228284626-75c309c6-5889-401c-8969-652292afb357.png)
Doesn't seem to work for me

Reply:
Yes it is not working